include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -1167,34 +1166,15 @@ std::optional<SmallVector<SmallVector<unsigned>>>
 emitOffsetForLayoutUsingLinearLayouts(Attribute layout, RankedTensorType type);
 
 inline SmallVector<SmallVector<unsigned>>
-emitOffsetForLayout(Attribute layout, RankedTensorType type,
-                    bool allowLLs /*= true*/) {
-  if (allowLLs) {
-    std::optional<SmallVector<SmallVector<unsigned>>> llOffsets =
-        emitOffsetForLayoutUsingLinearLayouts(layout, type);
-    if (llOffsets.has_value())
-      return *llOffsets;
-  }
-
-  if (auto blockedLayout = dyn_cast<BlockedEncodingAttr>(layout))
-    return emitOffsetForBlockedLayout(blockedLayout, type);
-  if (auto mmaLayout = dyn_cast<NvidiaMmaEncodingAttr>(layout)) {
-    if (mmaLayout.isVolta())
-      return emitOffsetForMmaLayoutV1(mmaLayout, type);
-    if (mmaLayout.isAmpere())
-      return emitOffsetForMmaLayoutV2(mmaLayout, type);
-    if (mmaLayout.isHopper())
-      return emitOffsetForMmaLayoutV3(mmaLayout, type);
-  }
-  if (auto mfmaLayout = mlir::dyn_cast<AMDMfmaEncodingAttr>(layout)) {
-    return emitOffsetForMfmaLayout(mfmaLayout, type);
-  }
-  if (auto wmmaLayout = mlir::dyn_cast<AMDWmmaEncodingAttr>(layout)) {
-    return emitOffsetForWmmaLayout(wmmaLayout, type);
-  }
-  if (auto sliceLayout = mlir::dyn_cast<SliceEncodingAttr>(layout))
-    return emitOffsetForSliceLayout(sliceLayout, type);
-  llvm_unreachable("unsupported emitOffsetForLayout");
+emitOffsetForLayout(Attribute layout, RankedTensorType type) {
+  std::optional<SmallVector<SmallVector<unsigned>>> llOffsets =
+      emitOffsetForLayoutUsingLinearLayouts(layout, type);
```

Comment:
I think it should be infallible, no need for returning `optional` here.  So essentially we replace emitOffsetForLayout with the LL version.

Reply:
In order to make it infallible, you need to make the function that converts a legacy layout to a linear layout infallible (i.e. doesn't return optional).